spec.emu

```diff
@@ -6,77 +6,178 @@
-      </emu-alg>
+      <p><ins>A <dfn id="dynamicimportstate-record" variants="DynamicImportState Records">DynamicImportState Record</dfn> is a Record that contains information about the loading process of a module graph. It's used to continue loading after a call to HostLoadImportedModule. Each DynamicImportState Record has the fields defined in <emu-xref href="#table-dynamicimportstate-record-fields"></emu-xref>:</ins></p>
+      <emu-table id="table-dynamicimportstate-record-fields" caption="DynamicImportState Record Fields">
+        <table>
+          <tr>
+            <th>
+              Field Name
+            </th>
+            <th>
+              Value Type
+            </th>
+            <th>
+              Meaning
+            </th>
+          </tr>
+          <tr>
+            <td>
+              <ins>[[PromiseCapability]]</ins>
+            </td>
+            <td>
+              <ins>a PromiseCapability Record</ins>
+            </td>
+            <td>
+              <ins>The promise to resolve when the dynamic import finishes.</ins>
+            </td>
+          </tr>
+          <tr>
+            <td>
+              <ins>[[ImportAsModuleObject]]</ins>
+            </td>
+            <td>
+              <ins>a Boolean</ins>
+            </td>
+            <td>
+              <ins>It is true if the <code>module: true</code> option was specified.</ins>
+            </td>
+          </tr>
+        </table>
+      </emu-table>
 
-      <emu-grammar><ins>ImportCall : `import` `(` AssignmentExpression `,` AssignmentExpression `)`</ins></emu-grammar>
-      <emu-alg>
-        1. <ins>Return ? EvaluateImportCall(the first |AssignmentExpression|, the second |AssignmentExpression|).</ins>
-      </emu-alg>
+      <emu-clause id="sec-ContinueDynamicImport" type="abstract operation">
+        <h1>
+          ContinueDynamicImport (
+            <del>_promiseCapability_: a PromiseCapability Record,</del>
+            <ins>_state_: a DynamicImportState Record,</ins>
+            _moduleCompletion_: either a normal completion containing a Module Record or a throw completion,
+          ): ~unused~
+        </h1>
+        <dl class="header">
+          <dt>description</dt>
+          <dd>It completes the process of a dynamic import originally started by an <emu-xref href="#sec-import-calls">`import()`</emu-xref> call, resolving or rejecting the promise returned by that call as appropriate.</dd>
+        </dl>
+        <emu-alg>
+          1. <ins>Let _promiseCapability_ be _state_.[[PromiseCapability]].</ins>
+          1. If _moduleCompletion_ is an abrupt completion, then
+            1. Perform ! Call(_promiseCapability_.[[Reject]], *undefined*, &laquo; _moduleCompletion_.[[Value]] &raquo;).
+            1. Return ~unused~.
+          1. Let _module_ be _moduleCompletion_.[[Value]].
+          1. <ins>If _state_.[[ImportAsModuleObject]] is *true*, then</ins>
+            1. <ins>Let _moduleObject_ be GetModuleObject(_module_).</ins>
```

Comment:
Rather than `GetModuleObject`, the reflection for WebAssembly is the direct `WebAssembly.Module` object which is being exposed behind a host hook integration. To align this with JS should be to align the `ModuleSource` object and not the `Module` instance object.